teps/0059-skip-guarded-task-only.md

```diff
@@ -0,0 +1,276 @@
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+This TEP addresses skipping strategies to give users the flexibility to skip a single guarded `Task` only and unblock execution of its dependent `Tasks`.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this TEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a TEP within the wider Tekton community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+When [`WhenExpressions`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#guard-task-execution-using-whenexpressions) evaluate to `False`, the guarded [`Task`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#adding-tasks-to-the-pipeline) is skipped and its dependent `Tasks` are skipped as well while the rest of the [`Pipeline`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#pipelines) executes. Users need the flexibility to skip that guarded `Task` only and unblock the execution of the dependent `Tasks`.
+
+`Pipelines` are directed acyclic graphs where:
+- `Nodes` are `Tasks`
+- `Edges` are defined using ordering ([`runAfter`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#using-the-runafter-parameter)) and resource (e.g. [`Results`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#using-results)) dependencies
+- `Branches` are made up of `Nodes` that are connected by `Edges`
+
+Even though `WhenExpressions` are specified within `PipelineTasks`, they are currently *located* along the `Edges` between `Nodes` guarding `Branches` (similarly to [`Conditions`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#guard-task-execution-using-conditions)).
+
+Take this example:
+
+```
+tasks:
+  - name: previous-task
+    taskRef:
+      name: previous-task
+  - name: current-task
+    runAfter: [previous-task]
+    when:
+      - input: "foo"
+        operator: in
+        values: ["bar"]
+    taskRef:
+      name: current-task
+  - name: next-task
+    runAfter: [current-task]
+    taskRef:
+      name: next-task
+```
+The graph workflow is:
+
+```
+        previous-task          # executed
+             |
+          (guard)              # false         
+             |
+             v
+        current-task           # skipped
+             |
+             v
+         next-task             # skipped
+```
+This TEP aims to support this graph workflow as well:
+```
+        previous-task          # executed
+             |
+             v
+(guard) current-task           # false and skipped
+             |
+             v
+         next-task             # executed
+```
+The above workflow *locates* `WhenExpressions` within `Nodes`.
```

Comment:
I am not sure if I understand this statement. In this context `nodes` are `Task` CRDs, where are we relocating `WhenExpressions`,  in `Task` CRD? 🤔

Reply:
no no, I meant that in the graph visualization, it'd be located in the node as opposed to between nodes (will reword it)

Reply:
@pritidesai reworded it, please let me know what you think, thank you 😁

Reply:
yup, thanks @jerop for rewriting it 👍